packages/DynamicHyperlink/src/DynamicHyperlink.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import "./DynamicHyperlink.scss";
+
+export default function DynamicHyperlink({ onFetch }) {
+  const I18n = useI18n();
+
+  React.useEffect(() => {
+    function updateDynamicHyperlinks() {
+      const dynamicHyperlinks = Array.from(document.querySelectorAll("[data-dynamic-hyperlink]"));
```

Comment:
just pointing out that this assumes [the component is a singleton](https://overreacted.io/writing-resilient-components/#principle-3-no-component-is-a-singleton). Would it make sense for it to take in a parent element and then do `parent.querySelectorAll`?

Reply:
The other thing that I was thinking is what would happen if the link resides under a react tree, and this component which is at the top goes and modify the DOM. 

React would not know about this change, and then what would happen later if that component updated its tree, I guess will ignore the DOM and will put back an unstyled link?

This component is an interesting one.

Reply:
@alexzherdev Hmm I think we might learn more after it is implemented for testing.

@nahumzs The script includes a `MutationObserver`, so if the DOM changes, the script re-runs.